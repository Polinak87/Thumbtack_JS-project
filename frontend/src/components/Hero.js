import React from 'react';
import classNames from 'classnames';

export const aquamarine = 'is-primary';
export const blue = 'is-info';

export default function Hero({ className, text }) {
  const classes = () => {
    return classNames('hero', className);
  };

  return (
    <section className={classes()}>
      <div className="hero-body">
        <div className="container">
          <h1 className="title">{text}</h1>
        </div>
      </div>
    </section>
  );
}
