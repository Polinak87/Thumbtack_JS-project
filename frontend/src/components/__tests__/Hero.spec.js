import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Hero, {aquamarine} from '../Hero';

afterEach(cleanup);

describe('Hero', () => {
  it('should has text', () => {
    const text = 'Your inventory';
    const props = {
      className: aquamarine,
      text,
    }
    const { getByText } = render(<Hero {...props} />);
    expect(getByText(text).textContent).toBe(text);
  });

  it('renders correctly', () => {
    const text = 'Your inventory';
    const props = {
      className: aquamarine,
      text,
    }
    const { container } = render(<Hero {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});