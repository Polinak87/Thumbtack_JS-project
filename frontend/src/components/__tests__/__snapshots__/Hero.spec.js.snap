// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<section
  class="hero is-primary"
>
  <div
    class="hero-body"
  >
    <div
      class="container has-text-centered"
    >
      <h1
        class="title"
      >
        Your inventory
      </h1>
    </div>
  </div>
</section>
`;
